Переход на след EditText/на клик login по нажатию enter

sinhro200 opened the pull request:
Сдвиг кнопки login при открытии клавиатуры

sinhro200 committed:
Переход на след EditText/на клик login по нажатию enter
сдвиг кнопки при открытии клавиатуры

sinhro200 committed:
Поменял цвет editText'ов ввода пароля и логина

sinhro200 committed:
Реализован главный экран
 * Переключение по фрагментам с помошью Bottom nav view

dvysotskii commented on app/src/main/java/com/sinhro/memesapp_surf/ui/login/CustomLoadingButton.kt:
```diff
@@ -20,7 +20,7 @@ class CustomLoadingButton @JvmOverloads constructor(
 ) :
     RelativeLayout(context, attributeSet,defStyle,defStyleRes) {
-    lateinit var login_button: Button
+    lateinit var button: Button
```
login_button было правильное название для этой кнопки)

dvysotskii commented on app/src/main/java/com/sinhro/memesapp_surf/ui/login/CustomLoadingButton.kt:
```diff
@@ -61,35 +61,42 @@ class CustomLoadingButton @JvmOverloads constructor(
-        login_button = findViewById(R.id.log_in_btn_custom_button_view)
-        loading_iv = findViewById(R.id.loader_iv_custom_button_view)
+    override fun hasOnClickListeners(): Boolean {
+        return button.hasOnClickListeners()
```
что то много разных методов отвечающий за клик по кнопке!
Все можно попроще записать, login_btn.setOnClickListener{//код который необходимо выполнить при клике}

dvysotskii commented on app/src/main/java/com/sinhro/memesapp_surf/ui/login/LoginActivity.kt:
```diff
@@ -60,6 +66,35 @@ class LoginActivity : AppCompatActivity() {
+        }
+    }
+
+    fun hideKeyboardFrom(context: Context, view: View) {
```
лучше завести утилиту, KeyboardUtil, и вынести туда этот метод

dvysotskii commented on app/src/main/java/com/sinhro/memesapp_surf/ui/login/CustomLoadingButton.kt:
```diff
@@ -61,35 +61,42 @@ class CustomLoadingButton @JvmOverloads constructor(
-    fun setOnClickButton(onClickListener: OnClickListener) {
-        login_button.setOnClickListener(onClickListener)
+    override fun performClick(): Boolean {
+        return  if (button.isActivated) button.performClick()  else false
```
performClick() лучше не использовать! Любой клик через clickListener должен обрабатываться

dvysotskii commented on app/src/main/java/com/sinhro/memesapp_surf/ui/login/CustomLoadingButton.kt:
```diff
@@ -61,35 +61,42 @@ class CustomLoadingButton @JvmOverloads constructor(
-        login_button.text = buttonText
+        button.isEnabled = true
+        button.isActivated = true
+        button.visibility = View.VISIBLE
```
isVisibly есть экстеншен , он принимает true/false

sinhro200 committed:
Вынес textColor:White в тему всего приложения

sinhro200 committed:
Фрагмент со списком мемов
   * Сделана Логика запроса
   * Не совсем законченное UI

sinhro200 committed:
Удалил ненужный класс

sinhro200 committed:
Добавил свайп и loading icon при обновлении/загрузке мемов

sinhro200 committed:
Пикчи догрузил

sinhro200 committed:
Заменил все иконки на векторные

sinhro200 committed:
Доделал ячейку мема
 * добавлена возможность лайкнуть (сохранение в локальное хранилище и проверка на понравившиеся оттуда же)
  * допилено оформление ячейки мема
  * иконка обновления списка мемов

sinhro200 committed:
Сделала корректное обновление мемов
 * заменил кастомный RefreshListener на SwipeRefreshLayout

sinhro200 committed:
Доделал ячейку мема
 * добавлена возможность лайкнуть (сохранение в локальное хранилище и проверка на понравившиеся оттуда же)
  * допилено оформление ячейки мема
  * иконка обновления списка мемов

sinhro200 committed:
Починил разметку чтобы последний мем не обрезался

sinhro200 committed:
Настроил корректное отображение сообщения об ошибке при отсутствии интернета

sinhro200 committed:
Добавил активити для просмотра мема

sinhro200 committed:
по мелочи

sinhro200 committed:
toolbar сделал единственным в MainActivity, а из фрагментов его меняю

sinhro200 committed:
Доделал страницу просмотра мема
Корректную обработку favorite мемов(при выходе из активити в списке мемов обновляется значение isFavorite)
частично сделал обработку даты мема(без падежей)

sinhro200 committed:
Перенёс логику проверки, лайкнул ли пользователь мем в подробной активити, в onActivityResult

sinhro200 committed:
Сделал базу данных, но для room нужен androidx, конфликтует с support
буду фиксить

sinhro200 committed:
Активити с добавлением мема,
ошибка с androidx не пофикшена

sinhro200 committed:
Мигрировал проект на androidx

sinhro200 committed:
Рефактор классов в UI

sinhro200 committed:
Активити с созданием мема + использование ранее созданной базы данных для сохранения
   * Некорректное отображение вкладки при выходе из Activity с созданием мема

sinhro200 committed:
Убрал ненужный try catch

sinhro200 committed:
Сделал профиль, пока не настроен список созданных мемов

sinhro200 committed:
Рабочий профиль, со списком своих мемов из бд

sinhro200 committed:
Поправил почему-то сломашшийся выбор фото из памяти

sinhro200 committed:
Выставил лаунчер в приложении на сплэш
